modules/bwa.nf

```diff
@@ -0,0 +1,56 @@
+process bwa_index {
+  label 'bwa'
+  
+  input:
+    path(fasta)
+  
+  output:
+    path(bwa) , emit: index
+  
+  script:
+  """
+  mkdir bwa
+  bwa \\
+    index \\
+    -p bwa/db \\
+    $fasta
+  """
+  
+  stub:
+  """
+  mkdir bwa
+  
+  touch bwa/db.amb
+  touch bwa/db.ann
+  touch bwa/db.bwt
+  touch bwa/db.pac
+  touch bwa/db.sa
+  """
+}
+
+process bwa {
+  label 'bwa'
+
+  input: 
+  tuple val(name), path(input)
+  path(db_index)
+  path(db)
+
+
+  output:
+  tuple val(name), val('raw'), path("${name}.bam"), emit: bam // input just for naming
+
+  script:
+  """
+  INDEX=`find -L ./ -name "*.amb" | sed 's/\\.amb\$//'`
+  bwa mem \\
+    -t $task.cpus \\
+    \$INDEX \\
+    $input \\
```

Comment:
`bwa mem` does automatically take care of single-end or paired-end read input, or? So you just pass `$input` and then this will be either a SE FASTQ or two PE FASTQs, right?

Reply:
yes, for single-end, it would be one fastq file; for paired-end, `$input` would be `foo_1.fastq foo_2.fastq` (space-separated)